src/Microsoft.Health.Extensions.DependencyInjection/ServiceCollectionExtensions.cs

```diff
@@ -23,12 +23,24 @@ public static class ServiceCollectionExtensions
         /// <param name="assembly">The assembly.</param>
         /// <param name="constructorParams">An array of constructor parameters that can be injected into IStartupModule on creation.</param>
         public static void RegisterAssemblyModules(this IServiceCollection collection, Assembly assembly, params object[] constructorParams)
+        {
+            collection.RegisterAssemblyModules(assembly: assembly, moduleTypeFilter: null, constructorParams: constructorParams);
+        }
+
+        /// <summary>
+        /// Registers assembly modules.
+        /// </summary>
+        /// <param name="collection">The collection.</param>
+        /// <param name="assembly">The assembly.</param>
+        /// <param name="moduleTypeFilter">Filter out modules that are not be registered.</param>
+        /// <param name="constructorParams">An array of constructor parameters that can be injected into IStartupModule on creation.</param>
+        public static void RegisterAssemblyModules(this IServiceCollection collection, Assembly assembly, Func<Type, bool> moduleTypeFilter, params object[] constructorParams)
```

Comment:
nit:
```suggestion
        public static void RegisterAssemblyModules(this IServiceCollection collection, Assembly assembly, Predicate<Type> moduleTypeFilter, params object[] constructorParams)
```

<br />
Another thought, from the description of your use-case, did you consider something like this instead?

`public static void RegisterModule(this IServiceCollection collection, IStartupModule module)`?

Reply:
@brendankowitz  thanks Brendan. I'm thinking to organize modules into different namespaces, one for DicomFunctions, the other for DicomWebServer. Would that make sense to you?